### PR TITLE
[test] Use LLM API for Nemotron-H correctness test

### PR DESCRIPTION
**Pull request by tomeras91**
Nemotron-H correctness tests doesn't just check that the completions are as expected, but rather also validates that logprobs aren't too far from reference logprobs. Initially, the test was implemented using a manual generation for-loop, since the LLM API didn't support returning context and generation logprobs.

After PRs #4538 and #4819 added context and generation logits support for the LLM API, it can now be used to compare logprobs as well, removing the need for a manual generation loop. It is also better to use the LLM API in the unittest since that's the standard way to use a model in TRTLLM.

Therefore, this PR merges the old Nemotron-H correctness and LLM API unittests, such that now the LLM API is used in the Nemotron-H correctness test.

Also, the reference generation logprobs were updated, since turns out there was a bug in the position_ids in the old manual generation loop. This further emphasizes why, when possible, standard APIs should be used instead of custom code.

### CODE REVIEW

**Review by vegaluisjose — APPROVED** [review]
LGTM

**Review by omera-nv — APPROVED** [review]
LGTM

**Review by amitz-nv — APPROVED** [review]
Use of get context & generation logits LGTM.
Note that logprobs support was merged this morning: https://github.com/NVIDIA/TensorRT-LLM/pull/4836 - using it can probably save some more code from this test
